br/pkg/lightning/backend/external/iter.go

```diff
@@ -198,6 +202,23 @@ func (i *mergeIter[T, R]) next() bool {
 	var zeroT T
 	i.curr = zeroT
 	if i.lastReaderIdx >= 0 {
+		cnt, ok := i.checkHotPointMap[i.lastReaderIdx]
+		if !ok {
+			i.checkHotPointMap[i.lastReaderIdx] = 1
+		} else {
+			i.checkHotPointMap[i.lastReaderIdx] = cnt + 1
+		}
+		i.checkHotPointCnt++
+
+		if i.checkHotPointCnt == 1000 {
+			i.checkHotPointCnt = 0
+			for idx, cnt := range i.checkHotPointMap {
+				(*i.readers[idx]).setReadMode(cnt > 500)
```

Comment:
seems byteReader is not concurrently used, we don;t need atomic.Bool to represent useConcurrentReader